src/runtime/virtcontainers/qemu_test.go

```diff
@@ -76,40 +76,112 @@ func TestQemuKernelParameters(t *testing.T) {
 }
 
 func TestQemuCreateVM(t *testing.T) {
-	qemuConfig := newQemuConfig()
 	assert := assert.New(t)
 
 	store, err := persist.GetDriver()
 	assert.NoError(err)
-	q := &qemu{
-		config: HypervisorConfig{
-			VMStorePath:  store.RunVMStoragePath(),
-			RunStorePath: store.RunStoragePath(),
-		},
-	}
-	sandbox := &Sandbox{
-		ctx: context.Background(),
-		id:  "testSandbox",
-		config: &SandboxConfig{
-			HypervisorConfig: qemuConfig,
-		},
-	}
 
 	// Create the hypervisor fake binary
 	testQemuPath := filepath.Join(testDir, testHypervisor)
 	_, err = os.Create(testQemuPath)
 	assert.NoError(err)
 
 	// Create parent dir path for hypervisor.json
-	parentDir := filepath.Join(store.RunStoragePath(), sandbox.id)
+	parentDir := filepath.Join(store.RunStoragePath(), "testSandbox")
 	assert.NoError(os.MkdirAll(parentDir, DirMode))
 
 	network, err := NewNetwork()
 	assert.NoError(err)
-	err = q.CreateVM(context.Background(), sandbox.id, network, &sandbox.config.HypervisorConfig)
-	assert.NoError(err)
+
+	type testData struct {
+		config      HypervisorConfig
+		expectError bool
+		configMatch bool
+	}
+
+	var data []testData
+
+	// create test configs with default settings
+	for i := 0; i < 18; i++ {
```

Comment:
Why is that a problem?

Reply:
Because it's fragile code that is hard-coding a magic number. That number and the `data[index]` block below this needs to be kept in-sync with this magic number.

In Kata, we try to avoid this as it makes maintenance more difficult.

Reply:
I could move the number to a standalone variable like "NumberOfTestConfigs" (in C I would just use a `#define`, not sure what's the go equivalent). The other altnernative is really to just have 18 lines repeating the same thing which I find silly.